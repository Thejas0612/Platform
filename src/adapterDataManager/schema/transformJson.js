import { template } from "lodash";

/**
 *
 * @param oldJson {object}
 * @param environmentVariables { {name:string, value:string}}
 * @return {object}
 */
export function transformJson(oldJson, environmentVariables) {
  const oldJsonString = JSON.stringify(oldJson);
  const compiled = template(oldJsonString);
  const newJsonString = compiled(environmentVariables);
  return JSON.parse(newJsonString);
}