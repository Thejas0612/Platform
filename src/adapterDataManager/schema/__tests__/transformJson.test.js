import { transformJson } from "../transformJson";

describe("transformJson", () => {
  it("when JSON contains environment variables, then replace all environment variables with actual values", () => {
    const environmentVariables = {
      VITE_API_URL: "http://localhost:5000"
    };

    const oldJson = {
      fields: [
        {
          name: "line-size",
          dataSourceUrl: "<%= VITE_API_URL %>/api/lookout/get_line_size?measurement_type=Flow"
        },
        {
          name: "pipe-schedule",
          dataSourceUrl: "<%= VITE_API_URL %}//api/lookout/get_pipe_schedule"
        }
      ]
    };

    const newJson = transformJson(oldJson, environmentVariables);

    expect(newJson).toMatchSnapshot();
  });

});

