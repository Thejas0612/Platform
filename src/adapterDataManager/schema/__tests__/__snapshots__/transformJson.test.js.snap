// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformJson when JSON contains environment variables, then replace all environment variables with actual values 1`] = `
{
  "fields": [
    {
      "dataSourceUrl": "http://localhost:5000/api/lookout/get_line_size?measurement_type=Flow",
      "name": "line-size",
    },
    {
      "dataSourceUrl": "<%= VITE_API_URL %}//api/lookout/get_pipe_schedule",
      "name": "pipe-schedule",
    },
  ],
}
`;
